test/Starcounter.Database.Extensions.IntegrationTests/DbStorage.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Starcounter.Database.Extensions.IntegrationTests
+{
+    class DbStorage
+    {
+        object _lock = new object();
+        Dictionary<int, object> _objects = new Dictionary<int, object>();
+
+        volatile int _counter = 0;
+
+        public ulong Insert(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            int id = ++_counter;
+            _objects.Add(id, obj);
+
+            return (ulong)id;
+        }
+
+        public ulong Delete(object obj)
+        {
+            if (obj == null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            int id = _objects.FirstOrDefault(t => object.ReferenceEquals(t.Value, obj)).Key;
```

Comment:
Could be replaced with:

```suggestion
            int id = _objects.First(t => object.ReferenceEquals(t.Value, obj)).Key;
```

Then the following `if (id == 0)` can be removed.

Reply:
No. That would raise a `InvalidOperationException`. I'm raising a `ArgumentOutOfRangeException`.

Reply:
Yes, it's a different exception.